### PR TITLE
Json api Error if the entity class is \Cake\ORM\Entity

### PR DESCRIPTION
**Pull request by dakota**
If the entity class for a repository is the default `\Cake\ORM\Entity` the the JSONAPI parser fails with a cryptic "Invalid schema" error. This causes the json api view class to throw a more verbose and descriptive error in that case to hint to the developer what they should be looking at.

### CODE REVIEW

**Comment by bravo-kernel** [comment]
@dakota is this PR still desired? If so I can try to fix once #520 gets merged.

**Comment by bravo-kernel** [comment]
One test failing on PHP 5.5

File: src/View/JsonApiView.php [hunk 1]
```diff
@@ -212,6 +213,10 @@ protected function _entitiesToNeoMerxSchema(array $repositories)
 
             $entityClass = $repository->entityClass();
 
+            if ($entityClass === Entity::class) {
+                throw new CrudException(sprintf('Entity classes must be the generic "%s" class for repository "%s"', $entityClass, $repositoryName));
```
Comment (ADmad) [inline src/View/JsonApiView.php]: ..must *not* be.. ?

**Comment by ADmad** [comment]
Test failing.

**Comment by bravo-kernel** [comment]
Fix in progress

**Comment by bravo-kernel** [comment]
A new release would be nice, I feel a bit hesitant to create one.

**Pull request merged**